src/libprojectM/CMakeLists.txt

```
@@ -0,0 +1,245 @@
+        "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/Renderer>"
+        "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/Renderer/hlslparser/src>"
+        "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/MilkdropPresetFactory>"
+        "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/NativePresetFactory>"
+        "$<INSTALL_INTERFACE:${PROJECTM_INCLUDE_DIR}>"
+        )
+
+target_link_libraries(projectM_shared
+        PUBLIC
+        OpenGL::OpenGL
+        OpenGL::GLX
+        ${CMAKE_DL_LIBS}
+        )
+
+if(ENABLE_THREADING)
+    target_link_libraries(projectM_shared
+            PUBLIC
+            Threads::Threads
+            )
+endif()
+
+enable_target_linker_flags_if_supported(projectM_shared PRIVATE --no-undefined)
+
+install(TARGETS projectM_static projectM_shared
+        EXPORT libprojectMTargets
+        LIBRARY DESTINATION "${PROJECTM_LIB_DIR}"
+        RUNTIME DESTINATION "${PROJECTM_LIB_DIR}"
+        ARCHIVE DESTINATION "${PROJECTM_LIB_DIR}"
+        COMPONENT Runtime
+        )
+
+install(FILES
+        Common.hpp
+        PCM.hpp
+        dlldefs.h
+        event.h
+        fatal.h
+        projectM.hpp
+        DESTINATION "${PROJECTM_INCLUDE_DIR}/libprojectM"
+        COMPONENT Devel
+        )
+
+
+
+# CMake target exports
+
+# For use from a local projectM build tree (without installing)
+export(TARGETS
+        MilkdropPresetFactory
+        NativePresetFactory
+        Renderer
+        hlslparser
+        SOIL2
+        projectM_static
+        projectM_shared
+        NAMESPACE libprojectM::
+        FILE projectM-exports.cmake
+        )
+
+export(PACKAGE libprojectM)
+
+
+
+# For use from an installed package (system install, vcpkg, homebrew etc.)
+include(CMakePackageConfigHelpers)
+
+write_basic_package_version_file(
+        "${CMAKE_CURRENT_BINARY_DIR}/libprojectM/libprojectMConfigVersion.cmake"
+        VERSION ${PROJECT_VERSION}
+        COMPATIBILITY AnyNewerVersion
+)
+
+configure_package_config_file(libprojectMConfig.cmake.in
+        "${CMAKE_CURRENT_BINARY_DIR}/libprojectM/libprojectMConfig.cmake"
+        INSTALL_DESTINATION "${PROJECTM_LIB_DIR}/cmake/libprojectM"
+        PATH_VARS PROJECTM_BIN_DIR PROJECTM_INCLUDE_DIR PROJECTM_DATADIR_PATH
+        )
+
+install(FILES
+        "${CMAKE_CURRENT_BINARY_DIR}/libprojectM/libprojectMConfigVersion.cmake"
+        "${CMAKE_CURRENT_BINARY_DIR}/libprojectM/libprojectMConfig.cmake"
+        DESTINATION "${PROJECTM_LIB_DIR}/cmake/libprojectM"
+        COMPONENT Devel
+        )
+
+install(EXPORT libprojectMTargets
+        FILE libprojectMTargets.cmake
+        NAMESPACE libprojectM::
+        DESTINATION "${PROJECTM_LIB_DIR}/cmake/libprojectM"
+        COMPONENT Devel
+        )
+
+
+
+# pkgconfig export
+macro(set_pkg_config_path varname path)
+    if(IS_ABSOLUTE "${path}")
+        set(${varname} "${path}")
+    else()
+        set(${varname} "\${prefix}/${path}")
+    endif()
+endmacro()
+
+set(PKGCONFIG_PREFIX "${CMAKE_INSTALL_PREFIX}")
```

Is it possible to just write
```suggestion
set(PKGCONFIG_PREFIX CMAKE_INSTALL_PREFIX)
```
?

---

Sadly no, only ‘if()‘ interprets variables directly. The ‘set()‘ command only takes the first argument as a variable name (also literally as a string) and then assigns everything after that as a string. The quotes are also required, otherwise spaces in a path can end up as a list separator, as CMake interprets unquoted spaces as such. So using

‘set(SOME_PATH /path/to/dir with/space)‘

will be stored as ‘/path/to/dir;with/space‘ in the variable. This variable weirdness is one of the things I despise most in CMake, especially the scoping mechanisms, but you get used to it 😁